### DISCUSSION TITLE
Endpoint to trigger tasks

### DISCUSSION THREAD

**Discussion by sdepablos** [cuebook/CueObserve#135] [Ideas]
Right now CueObserve is using Celery to schedule tasks (plus Redis to save the config). This requires having the system always up for the scheduler to work. In my case, I'd prefer to run this as an "scale to 0" application, either via Cloud Run or App Engine. To that effect, my idea would be to define to schedule in Google Cloud Scheduler, which will then trigger the recalculation, without the need to have a system always up. Which would be the API endpoint to trigger a task?

**Comment by sachinkbansal** [comment]
@sdepablos I'm with you on "scale to 0". Just wanted to understand the use case in detail.

At a high level, CueObserve has 2 modules - app with an underlying database and anomaly detection jobs. App uses celery to schedule anomaly detection jobs. The results of these jobs are stored in the database and are available in the app 24/7.

Would you want "scale to 0" for 
1. only anomaly detection jobs
2. app + jobs

If it's # 2, then the results won't be available for consumption in the app. We'll have to figure out an alternative way to make results available.

If it's # 1, then that's in our roadmap.

**Reply by sdepablos** [reply]
@sachinkbansal #1 sounds good enough, this way we can scale the anomaly detection part independently from the visualization part. #2 it's actually not that hard once you've split the app, as then you can simply host the app in a platform that allows scaling to 0, but after #1 it's not that important.
